File: SampleSolution.Core/Models/Entities/Base/Entity.cs
using System;

namespace SampleSolution.Core.Models.Entities.Base
{
    public class Entity<T> : IEntity<T> // T can be Long, String, Guid and others
    {
        public T Id { get; set; }

        public DateTime CreatedTime { get; set; }
        public DateTime LastUpdateTime { get; set; }
    }
}

File: Clients/SampleSolution.Web/wwwrootsrc/core/scripts/app/models/_base/Entity.ts
import {jsonProperty, Serializable} from 'ts-serializable';

export class Entity<T> extends Serializable {

    @jsonProperty(Number, String, void 0)
    public id?: T = void 0;

    @jsonProperty(Date)
    public createdTime: Date = new Date();

    @jsonProperty(Date)
    public lastUpdateTime: Date = new Date();
}
